updated the top navbar according to the new design, addresses #55

talpitoo opened the pull request:
## Preview

![Screenshot 2025-04-02 at 10-22-18 pkgdown template for cynkra packages • cynkratemplate](https://github.com/user-attachments/assets/755540cf-98de-4ab3-a1be-3605c77a9193)

talpitoo committed:
updated the top navbar according to the new design, addresses #55

maelle commented:
Thanks! :sparkles: On the main website, the huge vertical space above navbar items seems more useful because there's the "Get in touch" link. I'd suggest either adding that too, or making the vertical space smaller (but I guess making the logo rectangle would be blasphemy :grin: ).

christophsax commented:
I think the proposal looks pretty, even if the marketing seems a bit too agressive for my taste.

Making it smaller would be an easy step towards more modesty. We could still have a lovely footer, like ropensci, where we tell a bit more about us (and add some ascii).

krlmlr commented:
Can we keep a delicate blue square of the right size in the header, and zoom it to a full-size cynkra square on hover?

talpitoo committed:
smaller logo, footersquare, cynkra color variables

talpitoo committed:
assets/favicons clean-up

talpitoo committed:
linked the live cynkra logo, instead of inline SVG, favicon assets

talpitoo committed:
frutiger font path update

talpitoo committed:
footer ASCII

talpitoo commented:
@maelle @krlmlr @christophsax i have updated the template:
- smaller, 100x100 logo (on the main website we enlarge the logo to 140x140 at the md:768 breakpoint, we are not doing that here)
- same font, letter spacing in top navbar items
- all top navbar items are on the right side (to match the main website)
- no rounded corners anywhere
- updated the favicons (although i am not sure how to test it, i am assuming they are being copied to the right folders)
- footer ASCII
- updated the README.md regarding the 'frutiger' fonts and paths

I won't be more creative without consulting with the designer first.

Please note that since this is the first time i am using `R` and `pkgdown`, this was my workflow (i am not sure if that should be documented in the `README.md` or it is obvious for data analysts):
```
sudo apt update
sudo apt install r-base
sudo apt install pandoc
sudo apt install libcurl4-openssl-dev libssl-dev libxml2-dev
cd /home/tt/Documents/tt/jobs/cynkra/cynkratemplate/
R
```

```r
install.packages("pkgdown")
install.packages("devtools")
devtools::install_github("r-lib/pkgdown")
devtools::install_github("cynkra/cynkratemplate")
devtools::install()
pkgdown::build_site(preview = FALSE)
pkgdown::clean_site()
```

I still do not understand the whole context, i.e. how this would work with https://dm.cynkra.com/, do i have to update something more, etc.? Also i've seen at https://dm.cynkra.com/ that it has a search box in the top navbar, while in this template there is none. I am not sure whether i should style that as well (remove rounder corners) or this covers everything. `TL;DR` i might have missed some technical details due to my lack of understanding the technicalities in the R ecosystem.

Reade for review.

maelle commented:
Here's how it looks for dm. I installed cynkratemplate from this branch then rebuilt the dm website. So we will circulate this change by rebuilding all websites after this PR is merged. Thank you!!

![image](https://github.com/user-attachments/assets/5c972a56-d6fb-4ee8-bf15-4e342446b98f)

talpitoo commented:
@maelle that screenshot helps a lot (for context).

I would tweak `cynkratemplate` a bit more:
- using a wider `.container` so that packages that have a lot of navigation items in the top navbar have enough room
- vertically center the top navbar items with the search box

What commands should i execute and in which order to try out my local `cynkratemplate` on my local `dm`?

maelle commented:
For faster iterations, from the cynkratemplate package, `devtools::load_all()` and then if cynkratemplate and dm are in the same folder, `pkgdown::build_site("../dm")`. This way no need to _install_ cynkratemplate. I've just checked that it works.

talpitoo committed:
fluid top navbar and vertical center of the top navbar items

talpitoo commented:
I wasn't able to make it work locally but i managed to test the centering issue.

talpitoo committed:
fix 🍔 alignment

maelle commented:
Thank you! What are the next steps, merging and then my propagating the change to check there's no issue? cc @krlmlr @christophsax

talpitoo commented:
I assume you, @krlmlr and @christophsax should approve this new look first, then all the above :).

christophsax commented:
Do you have a screenshot, @maelle? 

Let's have a look at this with Jonas. The last screenshot looks quite mis-aligned to me, and subjectively quite a bit worse than default https://dplyr.tidyverse.org. 

A lot of it may be that it breaks to two lines, which may be mostly dm's fault that is has too much in navbar. It also happens on the current version.

<img width="1082" alt="image" src="https://github.com/user-attachments/assets/ce3748bf-5e0d-4e43-9084-db7fd04af18b" />

The logo itself looks pretty, I think.

maelle commented:
![image](https://github.com/user-attachments/assets/77f6d26e-04d6-4735-b9c9-fc31c7a6a665)

christophsax commented:
thanks pretty, thanks! 

Nitty picky: the version number is not aligned at the bottom with the rest of the text: 
<img width="722" alt="image" src="https://github.com/user-attachments/assets/87cb79e7-45d9-4ebb-883a-fe0fb43c71c1" />

But somehow on top.

Good for me! And the logo looks quite inviting to click on and come to our website.

talpitoo commented:
@maelle at `cynkratemplate` i set that all menu items in the top navbar should be on the right side. Should that be done at the dm package as well? Shouldn't that inherit it from `cynkratemplate`?

krlmlr commented:
A preview is now shown at https://cynkratemplate.cynkra.com/ when pushing to a branch that starts with `docs-`, as I did. I love it.

Is it intentional that the square and header is a bit smaller here than on the main website?

I'd expect the scrolling to behave like on the main website -- make the header disappear gradually, and not after scrolling the first few pixels.

krlmlr commented:
Minor nit: Ligatures in the code font.

![image](https://github.com/user-attachments/assets/dd5f6da9-bca4-4d5b-b28b-d1adb9173fb4)

I had a similar problem in VS Code, Chat suggested something along the following lines:

You're seeing **text ligatures** from the Iosevka font, and what's happening is that it's applying *typographic ligatures* like `fi` and `fl`—common in typesetting—but not necessarily desirable in programming.

You want to **disable these "standard" ligatures**, but keep **programming ligatures** (like `>=`, `===`, etc.).

### ✅ Solution

You can configure VS Code to **only enable specific ligatures**, rather than all of them.

---

### 🛠 Steps:

1. Open your VS Code settings (`Ctrl + ,` or `Cmd + ,` on Mac).
2. Search for `editor.fontLigatures`.
3. Change its value from `true` to this:

```json
"editor.fontLigatures": "'calt', 'liga', 'dlig'"
```

This keeps only certain ligature features enabled:
- `'liga'` = standard ligatures (like `fi`, `fl`)
- `'dlig'` = discretionary ligatures (like `>=`, `===`)
- `'calt'` = contextual alternates (often used in programming fonts for smoother ligature application)

But since `'liga'` enables `fi`, `fl`, you’ll want to **omit it**.

---

### ✅ Recommended setting for your case:

```json
"editor.fontLigatures": "'calt', 'dlig'"
```

That should:
- ✅ Keep programming ligatures (like `>=`)
- ❌ Disable typographic ligatures like `fi`, `fl`

---

Let me know if you're using a specific Iosevka variant (`Iosevka Term`, `Iosevka Fixed`, etc.)—some variants are designed **without** typographic ligatures at all, which might be a more straightforward fix.

talpitoo committed:
reverted the code font to avoid ligatures

talpitoo commented:
> Is it intentional that the square and header is a bit smaller here than on the main website?

Yes.
Initially, i wanted to recreate the main website's look 1:1 but that would entail a lot of hacks so i decided to go as close as possible with the existing tools/theming (Bootstrap 5). On the main website the logo is 100x100 on mobile and switches to 140x140 on desktop but i turned that off here based on @christophsax 's [comment](https://github.com/cynkra/cynkratemplate/pull/58#issuecomment-2766798576).

> I'd expect the scrolling to behave like on the main website -- make the header disappear gradually, and not after scrolling the first few pixels.

That is also not possible with the current 'light touch' approach. I would have to override both the markup and the JS, and that's only possible if i provide a new template from scratch. But that would make upgrading `pkgdown` impossible because we ended up with 'handmade' templates and would have to maintain it every time there is an update. Unless there is a consensus with @christophsax, i wouldn't go that route.

> Minor nit: Ligatures in the code font.

Thank you for noticing. This is actually a DESIGN BUG in Figma which (luckily) i forgot to implement on the main website. Jonas was using 'Overpass Mono' for monospace text which looks good in headings, etc. but never tested it on a `<code>` block. I say we ignore that design requirement from Figma (to use Overpass Mono for `<code>`) and revert to the old/existing ways, cc. @christophsax . I have just reverted this in my last push and double-checked the main website (mostly blog articles) to make sure it is not happening there neither.

Ready for review, along with https://github.com/cynkra/dm/pull/2280.

maelle commented:
@christophsax @krlmlr any further comment?

christophsax commented:
I haven't seen a latest version, but it all sounds very sensible.

krlmlr commented:
Have you seen <https://pkgdown.r-lib.org/articles/customise.html>, specifically the part about template packages? Can we not do whatever we want by overriding CSS and adding HTML bits, and still stay compatible with pkgdown 2.x.x? We certainly can control which version of pkgdown is used for rendering the documentation, if we need to, at least for a few years.

Re font: Can we not teach Overpass Mono to ignore old-style ligatures, like Chat suggested?

The most recent preview should be available in a few minutes at <https://cynkratemplate.cynkra.com/dev/>, but I need to update that manually if new changes come in. I'll take another look, and I'd like to do a second iteration soon after merging this.

talpitoo commented:
> Can we not do whatever we want by overriding CSS and adding HTML bits...

That's what we are doing right now, e.g. we can inject pieces of code into predefined slots but cannot change the skeleton itself (BS5). For that i would have to create a whole new theme, like https://github.com/r-lib/pkgdown/blob/main/inst/BS5/ and there leave everything intact and edit only the top navbar https://github.com/r-lib/pkgdown/blob/main/inst/BS5/templates/navbar.html to get the desired effect. That's changes the effort/scope/complexity/maintainability of the theming from S to XL. That's why i suggested not to go down that route.

> Re font: Can we not teach Overpass Mono to ignore old-style ligatures, like Chat suggested?

At first i got confused because you've shared a VSCode setting. We can actually 'teach' Overpass Mono with `font-variant-ligatures: none` CSS style but after trying it out, IMO looks weird:
![Screenshot from 2025-04-09 08-22-07](https://github.com/user-attachments/assets/1d949559-f457-4f97-92e5-642f2e384bc1)

In my opinion Overpass Mono looks great on some headings and the ASCII art but not as the default `<code>` font. You can try this out on the new blog: inspect element and change to font to 'Overpass Mono' in case you are curious.

Long replies but the TL;DR is that i won't tweak further the PR.

krlmlr commented:
I agree this PR is not the right venue to discuss extensive changes.

@christophsax: Are you okay with <https://cynkratemplate.cynkra.com/dev/>?

maelle committed:
undo config change + simplify config

krlmlr commented:
Closing in favor of #63, it's essentially the same plus two commits, but updates <https://cynkratemplate.cynkra.com/dev/> when pushed to.

talpitoo commented:
> Can we keep a delicate blue square of the right size in the header, and zoom it to a full-size cynkra square on hover?

> Warming up to @krlmlr's Idea of animate it to full size on hover. I love the effect on mobile, where the logo has exactly the same placement as on the website - so clicking changes everything else, not the logo. With full size on hover we could have the same effect on Desktop.

But having it grow to full size on hover would mean to push everything else around, to make the top navbar thicker (longer height), which would feel weird. Also, someone could just hover without clicking. For a seamless transition between cynkratemplate/cynkraweb on desktop is to keep the logo the same size (140x140) on desktop as well (but you've already said that feels too big, too aggressive marketing).

talpitoo commented:
> Not sure if possible. If complicated, let's forget it, the smaller logo is great as it is.

I can position the logo absolutely and grow above the top navbar but in a separate PR as it might be visually strange so we can evaluate before merging.

talpitoo commented:
I have added scaling to the logo in the other PR.
